pkg/kubelet/cm/container_manager.go

```diff
@@ -66,6 +67,16 @@ type ContainerManager interface {
 	// UpdateQOSCgroups performs housekeeping updates to ensure that the top
 	// level QoS containers have their desired state in a thread-safe way
 	UpdateQOSCgroups() error
+
+	// SetDevicePluginHandler sets the device plugin handler
+	SetDevicePluginHandler(DevicePluginHandler)
+
+	// GetDevicePluginHandler returns the DevicePluginHandler
+	GetDevicePluginHandler() DevicePluginHandler
+}
+
+type DevicePluginHandler interface {
+	Devices() map[string][]*pluginapi.Device
```

Comment:
> Why can't returning a list of devices be an API of the container manager?

I would rather have a loose coupling of the Device Plugin Handler interface rather than tight coupling.

> Consumers of container manager would only care about the opaque resource name and quantity right? Why is the pluginapi object being exposed?

Right now yes. However in the future it will make more sense to expose the Devices attributes.
Additionally I am not sure that formatting the Devices for a specific output should be done at this level, it increases coupling.